Source/Charts/Charts/PieRadarChartViewBase.swift

```diff
@@ -668,64 +668,40 @@ open class PieRadarChartViewBase: ChartViewBase
     
     private func sampleVelocity(touchLocation: CGPoint)
     {
-        let currentTime = CACurrentMediaTime()
-        
-        _velocitySamples.append(AngularVelocitySample(time: currentTime, angle: angleForPoint(x: touchLocation.x, y: touchLocation.y)))
-        
+        let currentSample: AngularVelocitySample = {
+            let time = CACurrentMediaTime()
+            let angle = angleForPoint(x: touchLocation.x, y: touchLocation.y)
+            return AngularVelocitySample(time: time, angle: angle)
+        }()
         // Remove samples older than our sample time - 1 seconds
-        var i = 0, count = _velocitySamples.count
-        while (i < count - 2)
```

Comment:
it's a bit obscure to say this change is equivalent. can you explain a little?
The main difference here is `while (i < count - 2)` so it seems the last one is not changed. but with the new change it will remove the last one if `$0.time < currentSample.time - 1`

Reply:
Oops! This is not supposed to be here, I will remove tonight. This is from #3830. Can you take a look at it? It's a good change.